src/components/ProgramLoader/Assembly.tsx

```diff
@@ -44,29 +44,16 @@ function assemblyFromInputProgram(initialState: InitialState, program: number[])
       // remove leading whitespace
       l = l.trim();
       // replace labels
-      l = l.replace(/: @(.+)$/, "@block$1:");
+      l = l.replace(/(: )?@(.+)$/, "@block$2$1");
+      // remove line number
+      l = l.replace(/^[0-9]+: /, "\t");
+      // fix ecalli comments?
```

Comment:
Could you please shed a little bit more light on why is it necessary? I mean, not only this line but whole `fixedLines`.  

Ideally would see it as the comment inside the code.

Reply:
Sure thing! Added some in-code comments.